clang/lib/AST/OpenMPClause.cpp

```diff
@@ -2903,7 +2903,10 @@ TargetOMPContext::TargetOMPContext(
     const FunctionDecl *CurrentFunctionDecl,
     ArrayRef<llvm::omp::TraitProperty> ConstructTraits)
     : OMPContext(ASTCtx.getLangOpts().OpenMPIsTargetDevice,
-                 ASTCtx.getTargetInfo().getTriple()),
+                 ASTCtx.getTargetInfo().getTriple(),
+                 ASTCtx.getLangOpts().OMPTargetTriples.empty()
+                     ? llvm::Triple()
+                     : ASTCtx.getLangOpts().OMPTargetTriples[0]),
```

Comment:
What if several target triples is available?

Reply:
As per my understanding, in the current implementation of OpenMP offloading, only the first non-host target specified via the `-fopenmp-targets` option is actively supported during offloading. Hence, I have considered only the first target triple.

Reply:
There is `--offload-arch` available to support multiple targets. Does it apply to host?

Reply:
From what I understand, the --offload-arch option is mostly for non-host devices like GPUs and doesn’t apply to the host. For the host, the architecture is determined by the default target triple. Since the host isn’t treated as an offloading target, --offload-arch doesn’t affect its code generation.